src/load_layout.c

```diff
@@ -52,6 +52,25 @@ struct focus_mapping {
 static TAILQ_HEAD(focus_mappings_head, focus_mapping) focus_mappings =
     TAILQ_HEAD_INITIALIZER(focus_mappings);
 
+static void apply_pending_marks(void) {
+    if (num_marks > 0) {
+        LOG("applying %d pending marks\n", num_marks);
+        for (int i = 0; i < num_marks; i++) {
+            Con *con = marks[i].con_to_be_marked;
+            char *mark = marks[i].mark;
+            if (con) {
+                DLOG("applying pending mark “%s” to con %p\n", mark, con);
+                con_mark(con, mark, MM_ADD);
+            }
+            free(mark);
+        }
+
+        FREE(marks);
+        num_marks = 0;
+        LOG("done applying pending marks\n");
+    }
+}
```

Comment:
No need for `LOG()`-level messages, `con_mark` also prints a log message.

```suggestion
static void apply_pending_marks(void) {
    for (int i = 0; i < num_marks; i++) {
        Con *con = marks[i].con_to_be_marked;
        char *mark = marks[i].mark;
        if (con) {
            con_mark(con, mark, MM_ADD);
        }
        free(mark);
    }

    FREE(marks);
    num_marks = 0;
}
```

Reply:
Done.